src/components/DocumentationTopic/PrimaryContent/DeclarationSource.vue

```diff
@@ -52,6 +51,111 @@ export default {
       required: false,
     },
   },
+  computed: {
+    formattedTokens: ({
+      language,
+      formattedSwiftTokens,
+      tokens,
+    }) => (language === Language.swift.key.api ? formattedSwiftTokens : tokens),
+    // Return a formatted version of the tokens array, with additional
+    // indentation whitespace to break parameters onto individual lines for
+    // improved readability and scanning of Swift functions/initializers.
+    //
+    // This is implemented in a single pass loop where text tokens are updated
+    // at certain key points to insert spaces and newlines so that each
+    // parameter of a multi-parameter function gets its own line.
+    //
+    // Ideally this should be implemented with a tool like SwiftFormat in the
+    // future with the raw text before it gets tokenized for syntax
+    // highlighting—however, this post-tokenization JavaScript logic should be
+    // an improvement until that kind of work can be integrated.
+    //
+    // @param {Array} tokens The original syntax tokens.
+    //   See `DeclarationToken.props`
+    // @return {Array} A formatted version of the original tokens.
+    //   See `DeclarationToken.props`
+    formattedSwiftTokens: ({ tokens }) => {
+      const numSpacesForIndent = 4; // maybe this could be configurable in the future
+      const indent = ' '.repeat(numSpacesForIndent);
+      let indentedParams = false;
+      const newTokens = [];
+      let i = 0;
+      let j = 1;
+      let openParenTokenIndex = null;
+      let openParenCharIndex = null;
+      let closeParenTokenIndex = null;
+      let closeParenCharIndex = null;
+      let numUnclosedParens = 0;
+
+      // loop through every declaration token
+      while (i < tokens.length) {
+        // keep track of the current token and the next one (if any)
+        const token = tokens[i];
+        const newToken = { ...token };
+        const nextToken = j < tokens.length ? tokens[j] : undefined;
+
+        // loop through the token text to look for "(" and ")" characters
+        // eslint-disable-next-line no-plusplus
+        for (let k = 0; k < (token.text || '').length; k++) {
```

Comment:
I think you could extract the `.length` invocation outside the `for`, so it does not call it for every iteration.

Reply:
70e4bafaba541d782be6245e0d5d70e65f9eb550